pydra/engine/task.py

```diff
@@ -348,10 +346,37 @@ def _command_args_single(self, state_ind, ind=None):
                 cpath = cdir.joinpath(lpath.name)
                 value = str(cpath)
             if f.type is bool:
+                if "argstr" in f.metadata:
+                    cmd_add.append(f.metadata["argstr"])
                 if value is not True:
                     break
             else:
-                cmd_add += ensure_list(value, tuple2list=True)
+                if "argstr" in f.metadata:
+                    argstr = f.metadata["argstr"]
+
+                    if argstr.endswith("..."):
+                        argstr = argstr.replace("...", "")
+                        if "sep" in f.metadata and isinstance(value, list):
+                            cmd_el_str = f.metadata["sep"].join(
+                                [argstr.format(**{f.name: val}) for val in value]
+                            )
+                        else:
+                            raise Exception("should we have ... without sep??")
+
+                    else:
+                        if "sep" in f.metadata and isinstance(value, list):
+                            cmd_el_str = f.metadata["sep"].join(
+                                [str(val) for val in value]
+                            )
+                        else:
+                            cmd_el_str = argstr.format(**{f.name: value})
+                    cmd_add += cmd_el_str.split(" ")
+                else:
```

Comment:
does this mean argstr is not mandatory?

Reply:
it is not

Reply:
in nipype1 if argstr is absent it won't get added to the command line. this allows including other inputs that may be required for building the command but will not be used to generate the command line. (e.g. terminal_output).

Reply:
ok, I'll make a not. I will rewrite it after hackathon, so can think about it. right now in my examples `argstr` is never required